app/policies/client_policy.rb

```diff
@@ -2,19 +2,25 @@ class ClientPolicy < ApplicationPolicy
   class Scope < ApplicationScope
     def resolve
       return all if superadmin?
-      resolve_owner if social_worker?
+      return resolve_owner if department_manager_or_social_worker?
+      none
     end
   end
 
-  alias_method :index?,             :superadmin_or_social_worker?
-  alias_method :new?,               :superadmin_or_social_worker?
-  alias_method :create?,            :superadmin_or_social_worker?
+  # controller action policies
+  alias_method :index?,  :volunteer_managing_user?
+  alias_method :new?,    :volunteer_managing_user?
+  alias_method :create?, :volunteer_managing_user?
 
-  alias_method :show?,              :superadmin_or_social_workers_record?
-  alias_method :edit?,              :superadmin_or_social_workers_record?
-  alias_method :update?,            :superadmin_or_social_workers_record?
+  alias_method :show?,   :superadmin_or_social_workers_record?
```

Comment:
Not sure, but since we want department managers to create assignments, shouldn't they be able to see their details in order to decide for an assignment?